### PR TITLE
No use of C11 threads on Windows/Mac

### PR DESCRIPTION
**Pull request by petervdonovan**
This branch is an attempt to address the failing Windows tests on `main`.

### CODE REVIEW

File: .github/scripts/check-diff.sh
Comment (lhstrh) [inline .github/scripts/check-diff.sh:1]: Whoops. That was my silly mistake.

**Comment by petervdonovan** [comment]
OK so now this PR is blocked by the fact that the PR that we just merged is rejected by AppleClang. I guess I jumped the gun in marking ready for review.

**Comment by lhstrh** [comment]
> OK so now this PR is blocked by the fact that the PR that we just merged is rejected by AppleClang. I guess I jumped the gun in marking ready for review.

Clang it!

**Review by lhstrh — APPROVED** [review]
These fixes look good to me! Apologies for the bugs I left in the check diff logic...

**Pull request merged**